Runtime/Behaviours/CinemachineConfiner2D.cs

```diff
@@ -0,0 +1,464 @@
+                return displacementCamera;
+            }
+
+            private Matrix4x4 S, R, T; // Scale, Rotation, Translation matrices converting camera to confiner space
+            private Matrix4x4 S_inverse, R_inverse; // Scale, Rotation matrices converting displacement to camera space
+            private void CalculateDeltaTransformationMatrix()
+            {
+                if (m_boundingShape2D.transform.hasChanged)
+                {
+                    Transform boundingShapeTransform = m_boundingShape2D.transform;
+                
+                    m_positionDelta = boundingShapeTransform.position;
+                
+                    m_rotationDelta = Quaternion.Inverse(m_boundingShapeRotation) * boundingShapeTransform.rotation;
+                
+                    Vector3 localScale = boundingShapeTransform.localScale;
+                    m_scaleDelta.x = Math.Abs(m_boundingShapeScale.x) < UnityVectorExtensions.Epsilon
+                        ? 0
+                        : localScale.x / m_boundingShapeScale.x;
+                    m_scaleDelta.y = Math.Abs(m_boundingShapeScale.y) < UnityVectorExtensions.Epsilon
+                        ? 0
+                        : localScale.y / m_boundingShapeScale.y;
+                    m_scaleDelta.z = Math.Abs(m_boundingShapeScale.z) < UnityVectorExtensions.Epsilon
+                        ? 0
+                        : localScale.z / m_boundingShapeScale.z;
+                
+                    S = Matrix4x4.identity;
+                    S.m00 = Math.Abs(m_scaleDelta.x) < UnityVectorExtensions.Epsilon ? 0 : 1f / m_scaleDelta.x;
+                    S.m11 = Math.Abs(m_scaleDelta.y) < UnityVectorExtensions.Epsilon ? 0 : 1f / m_scaleDelta.y;
+                    S.m22 = Math.Abs(m_scaleDelta.z) < UnityVectorExtensions.Epsilon ? 0 : 1f / m_scaleDelta.z;
+                    S_inverse = S.inverse;
+
+                    R = Matrix4x4.Rotate(Quaternion.Inverse(m_rotationDelta));
+                    R_inverse = Matrix4x4.Rotate(m_rotationDelta);
+                    
+                    T = Matrix4x4.identity;
+                    T.m03 = -m_positionDelta.x;
+                    T.m13 = -m_positionDelta.y;
+                    T.m23 = -m_positionDelta.z;
+                }
+            }
+
+            private void CalculateOffset()
+            {
+                var offset = m_boundingShape2D.offset;
+                var boundingShapeTransform = m_boundingShape2D.transform;
+                m_offset = new Vector3(
+                    offset.x * boundingShapeTransform.localScale.x,
+                    offset.y * boundingShapeTransform.localScale.y,
+                    0
+                );
+                m_offset = boundingShapeTransform.rotation * m_offset;
+            }
+        }
+
+        private void OnValidate()
+        {
+            m_Damping = Mathf.Max(0, m_Damping);
+            m_MaxOrthoSize = Mathf.Max(0, m_MaxOrthoSize);
+        }
+    }
```

Comment:
You should also add a Reset() method and set the default member values there, instead of specifying default values in the declarations.  It's called when you select "Reset" from the inspector popup, and when the object is initially created, so it covers all cases, unlike the inline default values.